### PR TITLE
Add birthdate field to userprofile

### PR DESCRIPTION
**Pull request by dtopal**
-user can change from preferences
-gym admins can enter birthdate when creating user

### CODE REVIEW

**Comment by rolandgeider** [comment]
Awesome! I'll take a closer look as soon as I can

**Comment by dtopal** [comment]
Thanks!  Sorry, I'm still relatively new at this.  I'm a bit lost as to why it's failing the check now but passing the check locally.

**Comment by rolandgeider** [comment]
It seems we need a newer version of setuptools, there's already PR for that #372

**Comment by dtopal** [comment]
Ok, I'll pull in the changes and see if that works.

**Comment by dtopal** [comment]
Should I clean up the commit history?

**Comment by rolandgeider** [comment]
No need to, I am also not a big fan of constantly rebasing. I mean, if you want to feel free, but unless there are _a lot_ of unecessary commits, you can just leave them

**Comment by rolandgeider** [comment]
I'm currently finishing the 1.8 release, when that's over I'll take a look at his (but the first impression is good!)

**Comment by dtopal** [comment]
Thanks!

File: wger/core/signals.py [hunk 1]
```diff
@@ -40,5 +43,17 @@ def create_user_cache(sender, instance, created, **kwargs):
...
+def set_user_age(sender, instance, **kwargs):
+    '''
+    Inputting or changing a user's birthdate will auto-update the
+    user's age as well if user's age has not been set
+    '''
+    if instance.age is None and instance.birthdate is not None:
+        today = datetime.date.today()
+        birthday = instance.birthdate
+        instance.age = (today.year - birthday.year
+                        - ((today.month, today.day) < (birthday.month, birthday.year)))
```
Comment (rolandgeider) [inline wger/core/signals.py]: Shouldn't this be ``(month, day) < (bday.month, bday. day)``?

File: wger/core/models.py [hunk 2]
```diff
@@ -207,6 +222,14 @@ class UserProfile(models.Model):
                        validators=[MinValueValidator(10), MaxValueValidator(100)])
     '''The user's age'''
 
+    birthdate = models.DateField(verbose_name=('Date of Birth'),
+                                 help_text=_('Please use the format: mm/dd/yyyy'),
```
Comment (rolandgeider) [inline wger/core/models.py]: Django should be clever enough to adapt the format to the current locale so if browse the application in German you'd enter ``dd.mm.yyyy``. Also if datefields are rendered as date input fields (I'm not 100% sure that they are), the browser will validate the format as well

File: wger/gym/forms.py [hunk 3]
```diff
@@ -65,6 +65,9 @@ class GymUserAddForm(GymUserPermisssionForm, UserPersonalInformationForm):
     Form used when adding a user to a gym
     '''
 
+    birthdate = forms.DateField(required=False,
+                                help_text=_("yyyy/mm/dd"))
```
Comment (rolandgeider) [inline wger/gym/forms.py]: Same as above, we can just remove the help text

File: wger/core/signals.py (hunk 1 above)
Comment (dtopal) [inline wger/core/signals.py]: Yes it should, thanks!

File: wger/core/models.py (hunk 2 above)
Comment (dtopal) [inline wger/core/models.py]: Ah, I see.  I'll get to work on the changes.

Comment (dtopal) [inline wger/core/models.py]: Sorry, I have a quick question.

Should I be using the Html5DateField from wger/utils/fields.py for the bday field in the model and the Html5FormDateField from wger/utils/widgets.py for the form field?

Comment (rolandgeider) [inline wger/core/models.py]: Better not, it seems there are still some browser compatibility issues with date input elements, just leave it as it is

**Comment by dtopal** [comment]
I think that's all the changes.  Let me know if I missed something or if there are any other changes you think would be good.

**Pull request merged**